oshi-core/src/main/java/oshi/util/Util.java

```diff
@@ -33,73 +37,137 @@
+	 * a licence key. (The identifier may not be unique as in one case hashcode
+	 * could be same for multiple values, and the result may differ based on whether
+	 * the program is running with sudo/root permission.) The identifier string is
+	 * based upon the processor serial number, vendor, processor identifier, and
+	 * total processor count.
+	 * 
+	 * @return A string containing four hyphen-delimited fields representing the
+	 *         processor; the first 3 are 32-bit hexadecimal values and the last one
+	 *         is an integer value.
+	 */
+	public static String getComputerIdentifier() {
+		SystemInfo systemInfo = new SystemInfo();
+		OperatingSystem operatingSystem = systemInfo.getOperatingSystem();
+		HardwareAbstractionLayer hardwareAbstractionLayer = systemInfo.getHardware();
+		CentralProcessor centralProcessor = hardwareAbstractionLayer.getProcessor();
+		ComputerSystem computerSystem = hardwareAbstractionLayer.getComputerSystem();
+
+		String vendor = operatingSystem.getManufacturer();
+		String processorSerialNumber = computerSystem.getSerialNumber();
+		String processorIdentifier = centralProcessor.getIdentifier();
+		int processors = centralProcessor.getLogicalProcessorCount();
+
+		String delimiter = "-";
+
+		return String.format("%08x", vendor.hashCode()) + delimiter
+				+ String.format("%08x", processorSerialNumber.hashCode()) + delimiter
+				+ String.format("%08x", processorIdentifier.hashCode()) + delimiter + processors;
+	}
+
+	/**
+	 * The function returns a string for VM identification, based on 2 checks so far:
+	 *  	1. MAC address
+	 *  	2. Computer model
+	 * The checks are done against two constants defined above. More can be added to
+	 * the constants as we encounter more types.
+	 * 
+	 * @return A string indicating if the machine is a VM or not. If yes, what type
+	 */
+	public static String identifyVM() {
+
+		SystemInfo si = new SystemInfo();
+		HardwareAbstractionLayer hw = si.getHardware();
+
+		// Try well known MAC addresses
+		NetworkIF[] nifs = hw.getNetworkIFs();
+		boolean isvm = false;
+		for (NetworkIF nif : nifs) {
+			String mac = nif.getMacaddr().substring(0, 8).toUpperCase();
+			if (vmMacAddressOUI.containsKey(mac)) {
+				isvm = true;
+				return "On a VM: " + vmMacAddressOUI.get(mac);
+			}
+		}
+
+		// Try well known models
+		if (!isvm) {
+			String model = hw.getComputerSystem().getModel();
+			for (String vm : vmModelArray) {
+				if (model.contains(vm)) {
+					isvm = true;
+					return "On a VM: " + vm;
+				}
+			}
+		}
+
+		// Check manufacturer and model, for now only Microsoft
+		if (!isvm) {
+			String manufacturer = hw.getComputerSystem().getManufacturer();
+			String model = hw.getComputerSystem().getModel();
+			if (manufacturer.equals("Microsoft Corporation") 
+				&& model.equals("Virtual Machine")) {
+					isvm = true;
+					return "On a VM: Microsoft Hyper-V";
+			}
+		}
+		return "Couldn't detect VM";
```

Comment:
Return empty string if no vm.  Update javadoc comment to explain this return value.